crates/api/src/user/get_multiple_users_freebusy.rs

```diff
@@ -172,20 +172,18 @@ impl GetMultipleFreeBusyUseCase {
         events: Vec<CalendarEvent>,
         timespan: &TimeSpan,
         calendars_lookup: &HashMap<String, &Calendar>,
-    ) -> VecDeque<EventInstance> {
-        events
-            .into_iter()
-            .map(|event| {
-                // TODO: to fix
-                #[allow(clippy::unwrap_used)]
-                let calendar = calendars_lookup
-                    .get(&event.calendar_id.to_string())
-                    .unwrap();
-                event.expand(Some(timespan), &calendar.settings)
-            })
-            .filter(|instances| !instances.is_empty())
-            .flat_map(|instances| CompatibleInstances::new(instances).inner())
-            .collect()
+    ) -> Result<VecDeque<EventInstance>, UseCaseError> {
+        let mut instances = VecDeque::new();
+        for event in &events {
+            let calendar = calendars_lookup
+                .get(&event.calendar_id.to_string())
+                .ok_or(UseCaseError::InternalError)?;
```

Comment:
Converted to declarative style with a normal loop so that `?` can be used directly